booster/tools/booster/Proxy.hs

```diff
@@ -429,14 +464,28 @@ respondEither cfg@ProxyConfig{statsVar, boosterState} booster kore req = case re
                     -- depth, in case we previously looped
                     Log.logInfoNS "proxy" . Text.pack $
                         "Booster " <> show boosterResult.reason <> " at " <> show boosterResult.depth
-                    logStats ExecuteM (time + bTime, koreTime)
-                    pure $
-                        Right $
-                            Execute
-                                boosterResult
-                                    { depth = currentDepth + boosterResult.depth
-                                    , logs = combineLogs [rpcLogs, boosterResult.logs]
-                                    }
+                    -- perform post-exec simplification
+                    postExecResult <-
+                        simplifyExecResult logSettings r._module def boosterResult
+                    case postExecResult of
+                        Left (nextState, newLogs) ->
+                            executionLoop
+                                logSettings
+                                def
+                                ( currentDepth + boosterResult.depth
+                                , time + bTime
+                                , koreTime
+                                , combineLogs $ rpcLogs : boosterResult.logs : newLogs
```

Comment:
If Kore gave as new simplification logs here, we would not emit them at `-l SimplifyJson`. If we want to be faithful, we need to replicate the code on lines 406-414:
```
                                    -- Kore may have produced simplification logs during rewriting. If so,
                                    -- output them Kore at "SimplifyJson" level, effectively
                                    -- appending them to Booster's traces. Erase terms from the traces.
                                    when (isJust koreResult.logs) $ do
                                        outputLogsAtLevel (Log.LevelOther "SimplifyJson")
                                            . map RPCLog.logEntryEraseTerms
                                            . filter isSimplificationLogEntry
                                            . fromJust $ koreResult.logs
```

to avoid code duplication, we should factor-out this into a function and put it into `simplifyExecRequest`.

However, I do not think that this is absolutely necessary to include this change in this PR, as here we are more concerned with correctness than the completeness of the simplification logs. As we agreed to strip the RPC simplification logs anyway, we will deal with this as part of https://github.com/runtimeverification/haskell-backend/issues/3801.

Reply:
I'll just quickly add those lines to `simplifyExecResult`, that is quick to do.

Reply:
I checked the code once again today, and I believe `simplifyExecResult` calls `simplifyExecuteState`, which calls `handleSimplify`, which already outputs the simplification logs from `kore` at `SimplifyJson` level. So unless there is a code path that escaped my attention, I don't think we need to add any logging here.